### PR TITLE
Add permission checks to the API

### PR DESCRIPTION
**Pull request by orzechdev**
It adds permission checks to the authorized API endpoints.

The question is: are the set of permissions appropriately selected for endpoints in this PR?

### CODE REVIEW

File: apps/checkout-app/pages/api/payment-provider-settings.ts [hunk 1]
```diff
@@ -24,4 +24,11 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
     return res.status(500).json({ error });
   }
 }
-export default allowCors(requireAuthorization(handler));
+export default allowCors(
+  requireAuthorization(handler, [
+    "HANDLE_PAYMENTS",
+    "HANDLE_CHECKOUTS",
+    "MANAGE_ORDERS",
+    "MANAGE_CHECKOUTS",
```
Comment (jwm0) [inline apps/checkout-app/pages/api/payment-provider-settings.ts:32]:
```suggestion
    "HANDLE_PAYMENTS",
```

File: apps/checkout-app/pages/api/set-payment-provider-settings.ts [hunk 2]
```diff
@@ -45,4 +45,11 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
     return res.status(500).json({ error });
   }
 }
-export default allowCors(requireAuthorization(handler));
+export default allowCors(
+  requireAuthorization(handler, [
+    "HANDLE_PAYMENTS",
+    "HANDLE_CHECKOUTS",
+    "MANAGE_ORDERS",
+    "MANAGE_CHECKOUTS",
```
Comment (jwm0) [inline apps/checkout-app/pages/api/set-payment-provider-settings.ts:53]:
```suggestion
    "HANDLE_PAYMENTS",
```

**Review by jwm0 — APPROVED** [review]

File: apps/checkout-app/backend/auth.ts [hunk 3]
```diff
@@ -80,11 +81,42 @@ export const isAuthenticated = async (req: NextApiRequest) => {
   return await jwtVerifier.verify(token);
 };
 
-export const isAuthorized = async (req: NextApiRequest) => {
+export const hasPermissions = (
+  tokenData?: JwtPayload,
+  permissions?: PermissionEnum[]
+) => {
+  if (!permissions?.length) {
```
Comment (bmigirl) [inline apps/checkout-app/backend/auth.ts:88]: I'm not sure I get this logic. The function is called `hasPermissions`, and we pass `permissions` to it, and then if `permissions` length === 0 we return `true` as `hasPermissions` equals `true`...?

File: apps/checkout-app/pages/api/payment-provider-settings.ts [hunk 4]
```diff
@@ -24,4 +24,4 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
     return res.status(500).json({ error });
   }
 }
-export default allowCors(requireAuthorization(handler));
+export default allowCors(requireAuthorization(handler, ["HANDLE_PAYMENTS"]));
```
Comment (bmigirl) [inline apps/checkout-app/pages/api/payment-provider-settings.ts:27]: Looking at how permissions are hardcoded here, as I understand they'll be passed down to requireAuthorization -> isAuthorized ->  hasPermissions -> and then this condition `if (!permissions?.length)` will always return the same value, is that correct?

File: apps/checkout-app/backend/auth.ts (hunk 3 above)
Comment (orzechdev) [reply apps/checkout-app/backend/auth.ts:88]: This function says if does it has all `permissions` in `user_permissions`. When length === 0 we can early return true;

Comment (orzechdev) [reply apps/checkout-app/backend/auth.ts]: I've updated naming

**Review by bmigirl — APPROVED** [review]

**Pull request merged**